commands/utils/packagehandlers/yarnpackagehandler.go

```diff
@@ -19,5 +33,50 @@ func (yarn *YarnPackageHandler) UpdateDependency(vulnDetails *utils.Vulnerabilit
 }
 
 func (yarn *YarnPackageHandler) updateDirectDependency(vulnDetails *utils.VulnerabilityDetails, extraArgs ...string) (err error) {
-	return yarn.CommonPackageHandler.UpdateDependency(vulnDetails, extraArgs...)
+	executableYarnVersion, err := getYarnVersion()
+	if err != nil {
+		return
+	}
+
+	var tmpDirToDelete string
+	isV2AndAbove := version.NewVersion(executableYarnVersion).Compare(yarnV2Version) <= 0
+
+	if isV2AndAbove {
+		vulnDetails.Technology.SetPackageInstallationCommand(yarnV2PackageUpdateCmd)
+	} else {
+		vulnDetails.Technology.SetPackageInstallationCommand(yarnV1PackageUpdateCmd)
+		tmpDirToDelete, err = fileutils.CreateTempDir()
```

Comment:
when we createdir or open file or do something that we need to close the next line should be `defer close()`
or `defer fileutils.RemoveTempDir(tmpDirToDelete)` in your case